src/native/libs/System.IO.Compression.Native/zlib_allocator_win.c

```diff
@@ -37,6 +37,45 @@
  * crash dump. We use a special code that's easy to search for in Watson.
  */
 
+BOOL IsMitigationDisabled()
+{
+    enum _MitigationEnablementTristate
+    {
+        MITIGATION_NOT_YET_QUERIED = 0,
+        MITIGATION_DISABLED = 1,
+        MITIGATION_ENABLED = 2 // really, anything other than 0 or 1
+    };
+    static long s_fMitigationEnablementState = MITIGATION_NOT_YET_QUERIED;
+
+    // If already initialized, return immediately.
+    // We don't need a volatile read here since the publish is performed with release semantics.
+    if (s_fMitigationEnablementState != MITIGATION_NOT_YET_QUERIED)
+    {
+        return (s_fMitigationEnablementState == MITIGATION_DISABLED);
+    }
+
+    // Initialize the tri-state now.
+    // It's ok for multiple threads to do this simultaneously. Only one thread will win.
+    // Valid env var values to disable mitigation: "true" and "1"
+    // All other env var values (or error) leaves mitigation enabled.
+    //
+    // Buffer needs to be large enough to hold null terminator, but returned cch does not include
+    // null terminator. Note *exclusive* bounds of 0 and buffer length on returned cch.
+    // Ref: https://learn.microsoft.com/windows/win32/api/winbase/nf-winbase-getenvironmentvariable
+    CHAR pchBuffer[5]; // enough to hold "true" and a terminator
+    DWORD dwEnvVarLength = GetEnvironmentVariableA("DOTNET_SYSTEM_IO_COMPRESSION_DISABLEZLIBMITIGATIONS", pchBuffer, _countof(pchBuffer));
```

Comment:
Should we try to have this variable declared as static? If its value is, say, `-1` (default), we call GetEnvironmentVariableA. That way we only call it once during the lifetime of the program. What do you think?

Reply:
Ok I see what you're doing: you're doing the safe interlocked exchange below. The equivalent check to what I'm suggesting is in line 52. Nevermind.